integration-cli/docker_cli_daemon_test.go

```diff
@@ -937,3 +955,99 @@ func TestDaemonRestartKillWait(t *testing.T) {
 
 	logDone("wait - wait on a stopped container doesn't timeout")
 }
+
+// TestHttpsInfo connects via two-way authenticated HTTPS to the info endpoint
+func TestHttpsInfo(t *testing.T) {
+	d := NewDaemon(t)
+	d.Start("--tlsverify", "--tlscacert", "fixtures/https/ca.pem", "--tlscert", "fixtures/https/server-cert.pem",
+		"--tlskey", "fixtures/https/server-key.pem", "-H", testDaemonHttpsAddr)
+	defer d.Stop()
+
+	cli := client.NewDockerCli(nil, ioutil.Discard, ioutil.Discard, "", testDaemonProto,
+		testDaemonHttpsAddr, getTlsConfig("client-cert.pem", "client-key.pem", t))
+
+	if msg := setTimeout("Reading command output time out", 10*time.Second, func() string {
```

Comment:
This `setTimeout` is actually pretty complicated compared to implementing manually.

```go
c := make(chan struct{})
go func() {
    d.Cmd(...)
    c<-struct{}{}
}()

select {
case <-c:
    // something
case <-time.After(duration):
    t.Fatal("timeout")
}
```